File: src/lib/models/simplified-addin-info.model.ts
export interface SimplifiedAddinInfoModel {
  name: string;
  vendorId: string;
  email: string;
  description: string;
  csharpProjectName: string;
  addinVersion: string;
  reasonForExport: string | null;
}

export function getEmptySimplifiedAddinInfo(): SimplifiedAddinInfoModel {
  return {
    name: "",
    vendorId: "",
    email: "",
    description: "",
    csharpProjectName: "",
    addinVersion: "0",
    reasonForExport: null,
  };
}

File: src-tauri/src/services/admin/addin_exporter/models/simplified_addin_info_model.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Serialize, Deserialize, Clone)]
#[serde(rename_all = "camelCase")]
pub struct SimplifiedAddinInfoModel {
    pub name: String,
    pub email: String,
    pub vendor_id: String,
    pub description: String,
    pub csharp_project_name: String,
    pub addin_version: String,

    pub reason_for_export: Option<String>,
}
